exasol/analytics/udf/communication/gather_operation.py

```diff
@@ -94,18 +109,25 @@ def _send_local_leader_message_to_multi_node_leader(self):
             local_position=local_position, value_frame=value_frame
         )
 
+    @property
```

Comment:
Property method `multi_node_communicator` checks the private attribute `_multi_node_communicator` and raises an exception when it is `None` to avoid code duplication.

Questions: 
* Q1) The names of the private attribute and the property differ on in the prefix `_` and potentially could be mixed up. Should we use different names that can easier be distinguished? 
* Q2) The property is public. Should it be private?

Reply:
See last push